standard/structs.md

```diff
@@ -502,6 +504,50 @@ Automatically implemented properties ([§15.7.4](classes.md#1574-automatically-i
 
 > *Note*: This access restriction means that constructors in structs containing automatically implemented properties often need an explicit constructor initializer where they would not otherwise need one, to satisfy the requirement of all fields being definitely assigned before any function member is invoked or the constructor returns. *end note*
 
+A *property_declaration* ([§15.7.1](classes.md#1571-general)) for an instance property in a *struct_declaration* may contain the *property_modifier* `readonly`. However, a static property shall not contain that modifier.
+
+It is a compile-time error to attempt to modify the state of an instance struct variable via a readonly property declared in that struct.
```

Comment:
From [MS Learn](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/readonly-instance-members):

> Readonly can be applied to property accessors to indicate that this will not be mutated in the accessor. The following examples have readonly setters because those accessors modify the state of member field, but do not modify the value of that member field.
> ```
> public readonly int Prop1
> {
>     get
>     {
>         return this._store["Prop1"];
>     }
>     set
>     {
>         this._store["Prop1"] = value;
>     }
> }
> ```

This example demonstrates a subtly that might easily be missed from the statement on line 589:

> It is a compile-time error to attempt to modify the state of an instance struct variable via a readonly property declared in that struct.

An example like (maybe exactly) this one should be included in the Standard with explanatory text to cover:

- The “indirect” state of a readonly only struct can be modified; and
- The the somewhat oxymoronic `read-only set` make sense in C#

Unless there are such explanations already here somewhere…